packages/main/src/Toolbar.ts

```diff
@@ -0,0 +1,590 @@
+
+import ToolbarDesign from "./types/ToolbarDesign.js";
+import ToolbarStyling from "./types/ToolbarStyling.js";
+import ToolbarAlign from "./types/ToolbarAlign.js";
+
+import ToolbarButton from "./ToolbarButton.js";
+import ToolbarSpacer from "./ToolbarSpacer.js";
+import ToolbarItem from "./ToolbarItem.js";
+import ToolbarSeparator from "./ToolbarSeparator.js";
+
+type ActionsWidthMap = {
+	[key: string]: number
+};
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ * The <code>ui5-toolbar</code> component is used to create a horizontal layout with actions.
+ * The items can be overflowing in a popover, when the space is not enough to show all of them.
+ *
+ * <h3>Keyboard Handling</h3>
+ * The <code>ui5-toolbar</code> provides advanced keyboard handling.
+ * <br>
+ * <ul>
+ * <li>The control is not interactive, but can contain of interactive elements </li>
+ * <li>[TAB] - iterates through elements</li>
+ * </ul>
+ * <br>
+ *
+ * <h3>ES6 Module Import</h3>
+ * <code>import "@ui5/webcomponents/dist/Toolbar";</code>
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webc.main.Toolbar
+ * @extends sap.ui.webc.base.UI5Element
+ * @tagname ui5-toolbar
+ * @public
+ * @since 1.16.0
+ */
+@customElement({
+	tag: "ui5-toolbar",
+	languageAware: true,
+	renderer: litRender,
+	template: ToolbarTemplate,
+	staticAreaTemplate: ToolbarPopoverTemplate,
+	styles: ToolbarCss,
+	staticAreaStyles: ToolbarPopoverCss,
+	dependencies: [
+		Popover,
+		Button,
+		ToolbarButton,
+		ToolbarSpacer,
+		ToolbarSeparator,
+	],
+})
+
+/**
+ * Fired whenever the width of the content changes.
+ * @param {Number} contentWidth the value of the new changed width.
+ * @event sap.ui.webc.main.Toolbar.prototype.contentWidthChange
+
+ * @public
+ */
+@event("content-width-change", {
+	detail: {
+		contentWidth: { type: Number },
+	},
+})
+/**
+ * Fired whenever the toolbar popup closes and opens.
+ * @param {Boolean} open indicates the state of the popup.
+ * @public
+ * @event sap.ui.webc.main.Toolbar.prototype.overflowToggle
+ */
+@event("overflow-toggle", {
+	detail: {
+		open: { type: Boolean },
+	},
+})
+
+class Toolbar extends UI5Element {
+	/**
+	 * Indicates if the Toolbar is in disabled state.
+	 * @type {boolean}
+	 * @public
+	 * @name sap.ui.webc.main.Toolbar.prototype.disabled
+	 */
+	 @property({ type: Boolean })
+	 disabled!: boolean;
+
+	/**
+	 * Defines the design of the toolbar background. It can be solid or transparent.
+	 * @type {sap.ui.webc.main.types.ToolbarDesign}
+	 * @public
+	 * @defaultvalue "Solid"
+	 * @name sap.ui.webc.main.Toolbar.prototype.design
+	 */
+	@property({ type: ToolbarDesign })
+	design!: string;
+
+	/**
+	 * Defines the styling of the toolbar. It can be standard or clear (no border applied).
+	 * @type {sap.ui.webc.main.types.ToolbarStyling}
+	 * @defaultvalue "Standard"
+	 * @name sap.ui.webc.main.Toolbar.prototype.styling
+	 * @public
+	 */
+	@property({ type: ToolbarStyling })
+	styling!: string;
```

Comment:
ToolbarStyling should be used instead of string for the TypeScript as well. Also, default value is not defined (only documented).